specutils/analysis/template_comparison.py

```diff
@@ -153,20 +151,27 @@ def template_match(observed_spectrum, spectral_templates,
         The index of the spectrum with the smallest chi2 in spectral templates.
     chi2_list : `list`
         A list with all chi2 values found for each template spectrum.
+    redshift : `None` or `float`
+        The value of the redshift that provides the smallest chi2.
     """
+    final_redshift = None
+
     if hasattr(spectral_templates, 'flux') and len(spectral_templates.flux.shape) == 1:
 
         # Account for redshift if provided
         chi2_list = []
         if redshift is not None:
-            _, redshifted_spectrum, chi2_list = template_redshift(observed_spectrum, spectral_templates,
-                                                               redshift=redshift)
-            spectral_templates = redshifted_spectrum
+            final_redshift, redshifted_spectrum, chi2, normalized_spectral_template, \
+                chi2_inner_list = \
+                template_redshift(observed_spectrum, spectral_templates, redshift=redshift)
 
-        normalized_spectral_template, chi2 = _chi_square_for_templates(
-            observed_spectrum, spectral_templates, resample_method)
+        else:
+            normalized_spectral_template, chi2 = _chi_square_for_templates(
+                observed_spectrum, spectral_templates, resample_method)
+
+        chi2_list.append(chi2)
 
-        return normalized_spectral_template, chi2, 0, chi2_list
+        return normalized_spectral_template, chi2, 0, chi2_list, final_redshift
```

Comment:
Whether you attached new value to the end like this or use `dict`, that is `specutils` maintainer's call. It will be a breaking change either way.